Plug-Ins/Dialog Editor/DITLItemView.swift

```diff
@@ -0,0 +1,410 @@
+		needsDisplay = true
+		if event.modifierFlags.contains(.shift) { // Multi-selection.
+			NotificationCenter.default.post(name: DITLDocumentView.selectionWillChangeNotification, object: superview)
+			selected = !selected
+			NotificationCenter.default.post(name: DITLDocumentView.selectionDidChangeNotification, object: superview)
+		} else if selected { // Drag/resize of selected items.
+			var knobIndex = 0
+			let pos = convert(event.locationInWindow, from: nil)
+			for knob in calculateKnobRects() {
+				if let knob = knob, knob.contains(pos) {
+					trackKnob(knobIndex, for: event)
+					return
+				}
+				knobIndex += 1
+			}
+			if (event.clickCount % 2) == 0 {
+				NotificationCenter.default.post(name: DITLDocumentView.itemDoubleClickedNotification, object: superview, userInfo: [DITLDocumentView.doubleClickedItemView: self])
+			} else {
+				trackDrag(for: event)
+			}
+		} else { // Select and possibly drag around an unselected item:
+			NotificationCenter.default.post(name: DITLDocumentView.selectionWillChangeNotification, object: superview)
+			selected = true
+			for itemView in superview?.subviews ?? [] {
+				if let itemView = itemView as? DITLItemView,
+				   itemView.selected, itemView != self {
+					itemView.selected = false
+					itemView.needsDisplay = true
+				}
+			}
+			NotificationCenter.default.post(name: DITLDocumentView.selectionDidChangeNotification, object: superview)
+			if (event.clickCount % 2) == 0 {
+				NotificationCenter.default.post(name: DITLDocumentView.itemDoubleClickedNotification, object: superview, userInfo: [DITLDocumentView.doubleClickedItemView: self])
+			} else {
+				trackDrag(for: event)
+			}
+		}
+	}
+	
+	override func draw(_ dirtyRect: NSRect) {
+		switch type {
+		case .userItem:
+			let fillColor = (NSColor.systemBlue.blended(withFraction: 0.4, of: NSColor.white) ?? NSColor.lightGray).withAlphaComponent(0.7)
+			let strokeColor = NSColor.systemBlue.blended(withFraction: 0.4, of: NSColor.black) ?? NSColor.black
+			fillColor.setFill()
+			strokeColor.setStroke()
+			NSBezierPath.fill(self.bounds)
+			NSBezierPath.stroke(self.bounds)
+			
+			title.draw(at: NSZeroPoint, withAttributes: [.foregroundColor: NSColor.systemBlue, .font: NSFontManager.shared.font(withFamily: "Silom", traits: [], weight: 0, size: 12.0)!])
+		case .button:
+			let fillColor = NSColor.white
+			let strokeColor = NSColor.black
+			fillColor.setFill()
+			strokeColor.setStroke()
+			let lozenge = NSBezierPath(roundedRect: self.bounds, xRadius: 8.0, yRadius: 8.0)
+			lozenge.fill()
+			lozenge.stroke()
+			
+			let ps = NSMutableParagraphStyle()
+			ps.alignment = .center
+			let attrs: [NSAttributedString.Key:Any] = [.foregroundColor: strokeColor, .paragraphStyle: ps, .font: NSFontManager.shared.font(withFamily: "Silom", traits: [], weight: 0, size: 12.0)!]
+			let measuredSize = title.size(withAttributes: attrs)
+			var textBox = self.bounds
+			textBox.origin.y -= (textBox.size.height - measuredSize.height) / 2
```

Comment:
Since the views are all flipped, should this be +=?
```suggestion
			textBox.origin.y += (textBox.size.height - measuredSize.height) / 2
```